sql/popularize.go

```diff
@@ -0,0 +1,49 @@
+package sql
+
+// Copyright 2019 The SQLFlow Authors. All rights reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import (
+	"bufio"
+	"os"
+)
+
+// Popularize reads SQL statements from the file named sqlfile in the
+// ./testdata directory, and runs each SQL statement with db.
+func Popularize(db *DB, sqlfile string) error {
+	f, e := os.Open(sqlfile)
+	if e != nil {
+		return e
+	}
+	defer f.Close()
+
+	onSemicolon := func(data []byte, atEOF bool) (advance int, token []byte, err error) {
+		for i := 0; i < len(data); i++ {
+			if data[i] == ';' {
+				return i + 1, data[:i], nil
+			}
+		}
+		return 0, nil, nil
+	}
+
+	scanner := bufio.NewScanner(f)
+	scanner.Split(onSemicolon)
```

Comment:
How if there is a statement in the file `sqlfile` which contains `;` in a string constant? For example,

```sql
INSERT INTO toutiao.train_processed VALUES ('news;music;video')
```

there are `;` in the inserted value.

Reply:
This function does not consider this case, I just move it out from the original file. I'll add some comments here because we only need to process some sample data files, if we want to put it into general use, we should probably update this function.

Reply:
If the only use of the function `Popularize` is in `cmd/sqlflowserver/`, it seems that we should move it there. The directory `sql` doesn't seem the right place for a hacky test-only function.